tools/topology/topology2/include/components/dts/dts_spk.conf

```diff
@@ -0,0 +1,148 @@
+Object.Base.data."dts_spk" {
```

Comment:
add comments for what's table used for?

Reply:
This table is the initial parameters to DTS library.

Reply:
This looks like a full-spec config, which should be set from external rather than by default.
Why don't we use an empty config for default similar to topology1?

Reply:
@johnylin76 Yes, this is the full config with default parameters for DTS library. To use empty config or not set the parameters lead to some problems here. Instead, to put the default parameters here works fine.  We'll still need to set the parameters by UCM afterward since it varies by project.

Reply:
The empty config I meant is the ABI-header-only blob, e.g. https://github.com/thesofproject/sof/blob/cavs2.5-001-drop-stable/tools/topology/topology1/m4/dts_codec_adapter.m4#L5
```
bytes "
0x53,0x4f,0x46,0x00,0x00,0x00,0x00,0x00,
0x00,0x00,0x00,0x00,0x00,0xa0,0x01,0x03,
0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00"
```
Have you tried this config and still led to problems?

I think we can set the full config here for developing purposes, but should not push this to upstream. Instead, a minimal and least effective config as default is preferred.

p.s. if the config above doesn't work, maybe try this instead:
```
bytes "
0x53,0x4f,0x46,0x00,0x00,0x00,0x00,0x00,
0x0c,0x00,0x00,0x00,0x00,0xa0,0x01,0x03,
0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
0x00,0x00,0x00,0x00,0x0c,0x00,0x00,0x00,
0x00,0x00,0x00,0x00"
```

which contains a 12-byte module config {paramId=0, paramSize=0xc, paramData=[0x0]}

Reply:
@johnylin76 
I tried both of them before. They didn't work. But I notice a minor difference here now that makes the second one work. I'll test it further and update it.
```
bytes "
0x53,0x4f,0x46,0x34,0x00,0x00,0x00,0x00,
0x0c,0x00,0x00,0x00,0x00,0xa0,0x01,0x03,
0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,
0x01,0x00,0x00,0x00,0x0c,0x00,0x00,0x00,
0x00,0x00,0x00,0x00"
'''

Reply:
@johnylin76 Updated the table now.

Reply:
The updated bytes will pass a param entry {paramId=1, paramSize=0xc, paramData=[0x0]} to DTS library. Given that paramId=1 is used in the full config, which should be the effective param ID for DTS library.

My previous idea is to pass {paramId=0, paramSize=0xc, paramData=[0x0]} which avoids the effective ID for DTS. However it sounds that errors are produced by paramId=0.

As long as DTS library has no side effect by setting {paramId=1, paramSize=0xc, paramData=[0x0]}, that looks good to me.